WalletWasabi.Gui/Controls/WalletExplorer/SendTabViewModel.cs

```diff
@@ -101,7 +101,7 @@ protected override async Task BuildTransaction(string password, PaymentIntent pa
 					}
 				}
 
-				IRelativeHttpClient httpClient = Global.Synchronizer.WasabiClientFactory.NewHttpClient(() => payjoinEndPointUri, isolateStream: false);
+				TorHttpClient httpClient = Global.Synchronizer.WasabiClientFactory.NewHttpClient(() => payjoinEndPointUri, isolateStream: false);
```

Comment:
This will force us to have `ClearnetHttpClient` implementation forever in `TorHttpClient`. In my mind, they are two HTTP client implementations:

* Clearnet HTTP client (based on .NET's [HttpClient](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-5.0))
* Tor HTTP client (based on Tor SOCKS5)

Using `TorHttpClient` here also makes it harder to have any other implementation easily (e.g. HTTP client with some proxy or whatever).

That's why I like the previous code better. It tells you exactly what happens: *You obtain HTTP client that is predefined with base URI and `isolateStream=false`*.

Reply:
What is worse, this complicates unit testing too. When you have an interface like `IHttpClient`, you can easily write a testing implementation that returns some pre-defined responses. So you can test CJ functionality on lower level than it is now easy.